src/utils/ucx/ucx_utils.h

```diff
@@ -137,44 +149,50 @@ class nixlUcxContext {
                    unsigned long num_workers, nixl_thread_sync_t sync_mode);
     ~nixlUcxContext();
 
-    static bool mtLevelIsSupproted(nixl_ucx_mt_t mt_type);
-
     /* Memory management */
     int memReg(void *addr, size_t size, nixlUcxMem &mem);
-    std::unique_ptr<char []> packRkey(nixlUcxMem &mem, size_t &size);
+    [[nodiscard]] std::string packRkey(nixlUcxMem &mem);
     void memDereg(nixlUcxMem &mem);
 
     friend class nixlUcxWorker;
 };
 
+[[nodiscard]] bool nixlUcxMtLevelIsSupported(const nixl_ucx_mt_t) noexcept;
```

Comment:
what is the benefit of moving it outside? Maybe still define it as a context method and cache returned attrs in the context, to avoid doing query every time?

Reply:
maybe make it a member of `nixlUcxEngine` then?

Reply:
ok, just thought that making it a property of either context or engine class would be more clear.
can be a separate func though